### ISSUE TITLE
node@0.12.x udp and pm2

### ISSUE THREAD

**Issue by skoranga** [Unitech/PM2#1204]
Hi 
Looks like pm2 is acting very weird on `node@0.12.x` when used in **cluster** mode. On hitting the udp code twice pm2 is orphaning the node process and going stall. This works perfectly fine on plain node.

Created this example to show the issue - https://github.com/skoranga/testudpapp

Any idea what's going on?

-sanjeev

**Comment by jshkurti** [comment]
Bug confirmed.
It looks like it comes from Node.js itself.
https://github.com/joyent/node/issues/9261

**Comment by jshkurti** [comment]
Thanks for reporting this, I'm going to try to find a workaround.

**Comment by jshkurti** [comment]
And it was fixed in joyent/node#v0.12 (https://github.com/joyent/node/pull/8643/files) but it will take a while till it's published.
We can't really do anything about it for now in PM2 :/

Here is a workaround given by one of the Node.js contributors :
```
var cluster = require('cluster');
var http = require('http');
var dgram = require('dgram');
var numCPUs = require('os').cpus().length;

if (cluster.isMaster) {
  // Fork workers.
  for (var i = 0; i < numCPUs; i++) {
    cluster.fork();
  }

} else {

  var server = http.createServer(function (request, response) {
    var client = dgram.createSocket("udp4");

    client.bind({exclusive: true}, function() {
      var message = new Buffer("Hello");
      client.send(message, 0, message.length, 80, "google.com", function(err, bytes) {
        client.close();
      });
      response.writeHead(200, {"Content-Type": "text/plain"});
      response.end("Hello World\n");
    });
  });

  server.listen(1337);
}
```

**Comment by skoranga** [comment]
thanks, the workaround works. :+1:
